lib/aptible/cli/subcommands/metric_drain.rb

```diff
@@ -0,0 +1,119 @@
+module Aptible
+  module CLI
+    module Subcommands
+      module MetricDrain
+        def self.included(thor)
+          thor.class_eval do
+            include Helpers::Token
+            include Helpers::Database
+            include Helpers::MetricDrain
+
+            desc 'metric_drain:list', 'List all Metric Drains'
+            option :environment
+            define_method 'metric_drain:list' do
+              Formatter.render(Renderer.current) do |root|
+                root.keyed_list('description') do |node|
+                  scoped_environments(options).each do |account|
+                    account.metric_drains.each do |drain|
+                      node.object do |n|
+                        ResourceFormatter.inject_metric_drain(n, drain, account)
+                      end
+                    end
+                  end
+                end
+              end
+            end
+
+            desc 'metric_drain:create:influxdb HANDLE '\
+                 '--db DATABASE_HANDLE --environment ENVIRONMENT',
+                 'Create an InfluxDB Metric Drain'
+            option :db, type: :string
+            option :environment
+
+            define_method 'metric_drain:create:influxdb' do |handle|
+              account = ensure_environment(options)
+              database = ensure_database(options)
+
+              opts = {
+                handle: handle,
+                database_id: database.id,
+                drain_type: :influxdb_database
+              }
+
+              create_metric_drain(account, opts)
+            end
+
+            desc 'metric_drain:create:influxdb:custom HANDLE '\
+                 '--username USERNAME --password PASSWORD ' \
+                 '--url URL_INCLUDING_PORT',
+                 'Create an InfluxDB Metric Drain'
+            option :db, type: :string
+            option :username, type: :string
+            option :password, type: :string
+            option :url, type: :string
+            option :db, type: :string
+            option :environment
+            define_method 'metric_drain:create:influxdb:custom' do |handle|
+              account = ensure_environment(options)
+
+              config = {
+                address: options[:url],
+                username: options[:username],
+                password: options[:password],
+                database: options[:db]
+              }
+              opts = {
+                handle: handle,
+                drain_configuration: config,
+                drain_type: :influxdb
+              }
+
+              create_metric_drain(account, opts)
+            end
+
+            desc 'metric_drain:create:datadog HANDLE '\
+                 '--api_key DATADOG_API_KEY --environment ENVIRONMENT',
+                 'Create a Datadog Metric Drain'
+            option :api_key, type: :string
+            option :custom_url, type: :string
```

Comment:
A --site option would be much easier to use for customers, if they want to specify the site name that DD uses rather than have to know how to specify the full and correct API URL https://docs.datadoghq.com/getting_started/site/